src/app/design/frontend/base/default/template/payment/form/omisecc.phtml

```diff
@@ -1,77 +1,93 @@
+    // Hook Payment.save function by wrapping Omise logic on top of the 'Payment.prototype.save'
+    Payment.prototype.save = Payment.prototype.save.wrap(function(superMethod) {
+        var omisePayment = document.getElementById("p_method_omise_gateway");
 
-                    formValidation.css({'opacity': 0, 'display': 'none'});
-
-                    // Send a request to create a token
-                    // then trigger the callback function once a response is received from Omise.
-                    // * Note that the response could be an error and this needs to be handled
-                    // * within the callback.
-                    Omise.createToken("card", card, function (statusCode, response) {
-                        // If has an error (can not create a card's token).
-                        if (response.object == "error") {
-                            // Display an error message - "Omise Response: "+response.message.
-                            formValidation.css({'opacity': 1, 'display': 'block'});
-                            formValidationMsg.html("Omise Response: "+response.message);
-
-                            jQuery("#payment-buttons-container").find("button").prop("disabled", false);
-                        } else if (typeof response.card != 'undefined' && !response.card.security_code_check) {
-                            // Display an error message - "Omise Response: Card authorization failure.".
-                            formValidation.css({'opacity': 1, 'display': 'block'});
-                            formValidationMsg.html("Omise Response: Card authorization failure.");
-
-                            jQuery("#payment-buttons-container").find("button").prop("disabled", false);
-                        } else {
-                            form.find("[data-omise=holder_name]").prop("disabled", true);
-                            form.find("[data-omise=number]").prop("disabled", true);
-                            form.find("[data-omise=expiration_month]").prop("disabled", true);
-                            form.find("[data-omise=expiration_year]").prop("disabled", true);
-                            form.find("[data-omise=security_code]").prop("disabled", true);
-
-                            form.find("[data-omise=omise_token]").val(response.id);
-
-                            return superMethod();
-                        };
-                    });
+        if (! omisePayment.checked) {
+            return superMethod();
+        }
 
+        var validator = new Validation("co-payment-form");
+
+        if (validator.validate()) {
+            checkout.setLoadWaiting('payment');
+
+            getScript("https://cdn.omise.co/omise.min.js.gz", function() {
+                // Set Omise key
+                Omise.setPublicKey("<?php echo $this->getOmiseKeys('public_key'); ?>");
+
+                var form                  = document.getElementById("payment_form_omise_gateway"),
```

Comment:
A variable, form, was defined here but I found no any place using it.
Could you please guide me where it is used?